internal/cluster/cluster.go

```diff
@@ -998,6 +998,10 @@ func (m Manager) PermanentClustersDeletion(ctx context.Context, olderThan strfmt
 			deleteFromDB = false
 			m.log.WithError(err).Warnf("Failed deleting s3 manifests of cluster %s", c.ID.String())
 		}
+		if _, err := objectHandler.DeleteObject(ctx, c.ID.String()); err != nil {
```

Comment:
s3 doesn't have a directory what will happen there?

Reply:
When I inspected the structure on s3 (scality impl) I did noticed there is a directory for each cluster, e.g. `62529545-aad1-4739-8977-180d825245b3/`:

```
→ aws s3 ls --endpoint-url=http://localhost:8000 test
                           PRE 62529545-aad1-4739-8977-180d825245b3/
2021-06-20 11:38:54  957349888 discovery-image-62529545-aad1-4739-8977-180d825245b3.iso
2021-06-20 11:36:07         13 minimal_templates_version.json
```
and its content:
```
→ aws s3 ls --endpoint-url=http://localhost:8000 test/62529545-aad1-4739-8977-180d825245b3/
2021-06-20 11:37:56       8349 discovery.ign
```
The GC's worker logs indicates the folder is being deleted:
```
time="2021-06-20T09:32:27Z" level=info msg="Permanently deleting cluster 62529545-aad1-4739-8977-180d825245b3 that was de-registered before 2021-06-20T09:31:27.205Z" func=github.com/openshift/assisted-service/internal/cluster.Manager.PermanentClustersDeletion file="/go/src/github.com/openshift/origin/internal/cluster/cluster.go:988" pkg=cluster-state
time="2021-06-20T09:32:27Z" level=info msg="Listing objects by with prefix 62529545-aad1-4739-8977-180d825245b3/" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*S3Client).ListObjectsByPrefix" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/client.go:459" go-id=148 request_id=
time="2021-06-20T09:32:27Z" level=info msg="Deleting object 62529545-aad1-4739-8977-180d825245b3/discovery.ign from test" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*S3Client).DeleteObject" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/client.go:312" go-id=148 request_id=
time="2021-06-20T09:32:27Z" level=info msg="Deleted object 62529545-aad1-4739-8977-180d825245b3/discovery.ign from bucket test" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*S3Client).DeleteObject" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/client.go:328" go-id=148 request_id=
time="2021-06-20T09:32:27Z" level=info msg="Listing objects by with prefix 62529545-aad1-4739-8977-180d825245b3/logs/" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*S3Client).ListObjectsByPrefix" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/client.go:459" go-id=148 request_id=
time="2021-06-20T09:32:27Z" level=info msg="Listing objects by with prefix 62529545-aad1-4739-8977-180d825245b3/manifests/" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*S3Client).ListObjectsByPrefix" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/client.go:459" go-id=148 request_id=
time="2021-06-20T09:32:27Z" level=info msg="Deleting object 62529545-aad1-4739-8977-180d825245b3 from test" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*S3Client).DeleteObject" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/client.go:312" go-id=148 request_id=
time="2021-06-20T09:32:27Z" level=info msg="Deleted object 62529545-aad1-4739-8977-180d825245b3 from bucket test" func="github.com/openshift/assisted-service/pkg/s3wrapper.(*S3Client).DeleteObject" file="/go/src/github.com/openshift/origin/pkg/s3wrapper/client.go:328" go-id=148 request_id=
```
And eventually the bucket doesn't contain that folder:
```
→ aws s3 ls --endpoint-url=http://localhost:8000 test
2021-06-20 11:36:07         13 minimal_templates_version.json
```

Reply:
Strange, i would expect to get an error, there is no directory concept with object store.
But ok.